tests/auth/test_auth.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime, timedelta, timezone
+
+import pytest
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core import mail
+
+from django.test import Client
+from django.urls import reverse
+from django.utils.timezone import make_naive
+
+from auth.models import EmailToken
+from auth.service import create_token, validate_from_token
+
+
+def client():
+    return Client()
+
+
+def test_create_token_function(db, data):
+    uuid = "a603ae0a-4188-4098-99ca-3b853642c1c7"
+    today = datetime.now(timezone.utc)
+    token = create_token(data.niko, "newemail@gmail.com", uuid, today=today)
+    email_token = EmailToken.objects.get(token=token)
+
+    assert token == "a603ae0a-4188-4098-99ca-3b853642c1c7"
+    assert str(email_token.token) == "a603ae0a-4188-4098-99ca-3b853642c1c7"
+    assert today + timedelta(settings.EMAIL_ACTIVATION_DAYS) == email_token.expire_at
+
+
+def test_user_change_email_e2e(client, data):
+    _login_client(client, data)
+
+    new_email = "test@test.com"
+    _change_client_email(client, new_email)
+
+    email_token_query = EmailToken.objects.all()
+    assert len(email_token_query) == 1
+    assert email_token_query.first().token is not None
+
+    # test email notification sent
+    assert len(mail.outbox) == 1
+    assert "Activation de votre compte" in mail.outbox[0].subject
+    assert new_email in mail.outbox[0].recipients()
+    assert "Vous avez demandé à changer votre email" in mail.outbox[0].body
+
+
+def test_user_validate_email_change(db, data):
+    uuid = "a603ae0a-4188-4098-99ca-3b853642c1c7"
```

Comment:
Here and more generally, I'd love we use a single, consistent term for activation tokens; how about `activation_token`?